File: src/com/company/Customer.java
package com.company;

import java.io.Serializable;
import java.util.ArrayList;
import java.io.*;

public class Customer implements Serializable {

    private double totalOrderPrice;
    final private ArrayList pizzasOrdered = new ArrayList();
    private String name;
    private String phoneNumber;
    String customerFile = "customer.ser";

    public Customer(String name, String phoneNumber) {
        this.name = name;
        this.phoneNumber= phoneNumber;

    }

    public String getName() {
        return name;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }


    public double getTotalOrderPrice() {
        return totalOrderPrice;
    }

    public void setTotalOrderPrice(double totalOrderPrice) {
        this.totalOrderPrice = totalOrderPrice;
    }

    public void add(Pizza pizza){
        pizzasOrdered.add(pizza);
        this.totalOrderPrice = pizza.getPizzaTotal() + totalOrderPrice;
    }

}

File: src/PizzaPlus/Customer.kt
package PizzaPlus

import PizzaPlus.Order
import java.io.Serializable
import java.util.*

class Customer : Serializable {
    // Getters and Setters
    // Personal Information
    var phoneNumber: String? = null
    var firstName: String? = null
    var lastName: String? = null
    var streetAddress: String? = null
    var city: String? = null
    var state: String? = null
    var zipCode: String? = null

    // Order and Calculated Related Data
    var totalOrderPrice = 0.0
    var orders = ArrayList<Order>()
    fun newCustomerMenu() {  // Customer creation menu
        val scanner = Scanner(System.`in`)
        var profileDone = false
        var selection = 0
        while (!profileDone) {
            print("Enter First Name: ")
            firstName = scanner.nextLine()
            print("Enter Last Name: ")
            lastName = scanner.nextLine()
            print("Enter Street Address: ")
            streetAddress = scanner.nextLine()
            print("Enter City: ")
            city = scanner.nextLine()
            print("Enter State: ")
            state = scanner.nextLine()
            print("Enter City: ")
            zipCode = scanner.nextLine()
            print("Enter City: ")
            println("Is this information correct?")
            displayCustomerInfo()
            println("Enter 1 for yes, anything else to start over")
            selection = scanner.nextInt()
            if (selection == 1) {
                profileDone = true
            }
        }
        scanner.close()
    }

    fun displayCustomerInfo() {
        println("Name: " + firstName + " " + lastName)
        println("Address: " + streetAddress + ", " + city)
        println("State" + state + " Zip: " + zipCode)
    }

    fun add(order: Order) {
        orders.add(order)
        totalOrderPrice = order.total // function may not be necessary
    }
}